desktop/core/src/desktop/lib/thrift_util.py

```diff
@@ -338,6 +338,12 @@ def connect_to_thrift(conf):
   if conf.transport_mode == 'http':
     if conf.use_sasl and conf.mechanism != 'PLAIN':
       mode.set_kerberos_auth(service=conf.kerberos_principal)
+    elif USE_THRIFT_HTTP_JWT.get():
```

Comment:
Adding a short Mock Test? 

Handle None user and empty token more gracefully with an exception instead of crash?